viz/metrics-api/stat_summary.go

```diff
@@ -490,16 +490,30 @@ func buildTrafficSplitRequestLabels(req *pb.StatSummaryRequest) (labels model.La
 	return labels, groupBy
 }
 
+func buildTCPStatsRequestLabels(req *pb.StatSummaryRequest, reqLabels model.LabelSet) (labels model.LabelSet) {
+	switch req.Outbound.(type) {
+	case *pb.StatSummaryRequest_ToResource, *pb.StatSummaryRequest_FromResource:
+		// If TCP stats are queried from a resource to another one (i.e outbound -- from/to), then append peer='dst'
+		labels = reqLabels.Merge(promPeerLabel("dst"))
+
+	default:
+		// If TCP stats are not queried from a specific resource (i.e inbound -- no to/from), then append peer='src'
+		labels = reqLabels.Merge(promPeerLabel("src"))
+	}
+	return
+}
+
 func (s *grpcServer) getStatMetrics(ctx context.Context, req *pb.StatSummaryRequest, timeWindow string) (map[rKey]*pb.BasicStats, map[rKey]*pb.TcpStats, error) {
 	reqLabels, groupBy := buildRequestLabels(req)
 	promQueries := map[promType]string{
 		promRequests: reqQuery,
 	}
 
 	if req.TcpStats {
+		tcpLabels := buildTCPStatsRequestLabels(req, reqLabels)
 		promQueries[promTCPConnections] = tcpConnectionsQuery
-		promQueries[promTCPReadBytes] = tcpReadBytesQuery
-		promQueries[promTCPWriteBytes] = tcpWriteBytesQuery
+		promQueries[promTCPReadBytes] = fmt.Sprintf(tcpReadBytesQuery, tcpLabels.String(), timeWindow, groupBy.String())
```

Comment:
I noticed you left a comment in `prometheus.go` that explains why we don't have `fmt.Sprintf` on line 203. I think it would be good to also add a comment near here explaining why we have to format the string here and not in `prometheus.go`

Reply:
I was thinking of that too! I'll change this PR around a bit (see Alex's comment below). I'm not sure if this comment will still apply but will keep it in the back of my head; if after the change I can put it in, I will.

Thanks!